components/zcash_address/src/kind/unified.rs

```diff
@@ -279,12 +505,15 @@ pub(crate) mod private {
                         length
                     )));
                 }
-                let result = R::try_from((
-                    typecode,
-                    &buf[cursor.position() as usize..addr_end as usize],
-                ));
+                let data = &buf[cursor.position() as usize..addr_end as usize];
```

Comment:
```suggestion
                // The "as usize" casts cannot change the values, because both
                // cursor.position() and addr_end are unsigned values <= buf.len()
                // which is usize.
                let data = &buf[cursor.position() as usize..addr_end as usize];
```